packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm

```diff
@@ -279,11 +279,20 @@ id convertJSIValueToObjCObject(jsi::Runtime &runtime, const jsi::Value &value, s
                 return;
               }
 
-              NSDictionary *jsError = RCTJSErrorFromCodeMessageAndNSError(code, message, error);
-              reject->call([jsError](jsi::Runtime &rt, jsi::Function &jsFunction) {
-                jsFunction.call(rt, convertObjCObjectToJSIValue(rt, jsError));
+              NSDictionary *jsErrorDetails = RCTJSErrorFromCodeMessageAndNSError(code, message, error);
+              reject->call([jsErrorDetails](jsi::Runtime &rt, jsi::Function &jsFunction) {
+                // From JS documentation:
+                // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#cause an
+                // Error can be created with `new Error(message, option);`. the `option` param is a JS object with the
+                // `cause` property. Create a valid `option` object
+                NSDictionary<NSString *, id> *jsErrorOptions = @{@"cause" : jsErrorDetails};
+                auto jsiObjCError = convertObjCObjectToJSIValue(rt, jsErrorOptions);
+                NSString *message =
+                    jsErrorDetails[@"message"] ? jsErrorDetails[@"message"] : @"Unknown error from a native module";
+                auto jsError =
+                    rt.global().getPropertyAsFunction(rt, "Error").call(rt, [message UTF8String], jsiObjCError);
```

Comment:
Can you extract this logic to a helper, similarly to `convertNSExceptionToJSError` and `createJSRuntimeError`? There's a lot of overlap between these methods.